### ISSUE TITLE
update 4.0.0-x from 3.0.9 with error Unknown collation: 'utf8mb4'

### ISSUE THREAD

**Issue by Still4** [pingcap/tidb#17455]
kv is fine, db cannot start, with this error:

[2020/05/27 08:33:23.402 +00:00] [INFO] [printer.go:42] ["Welcome to TiDB."] ["Release Version"=v4.0.0-rc.2] [Edition=Community] ["Git Commit Hash"=dfbb1ff8b0a8d55893bcea652748c40b9a9b91ad] ["Git Branch"=heads/refs/tags/v4.0.0-rc.2] ["UTC Build Time"="2020-05-15 11:54:25"] [GoVersion=go1.13] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=v3.0.0-60965b006877ca7234adaced7890d7b029ed1306]
[2020/05/27 08:33:23.403 +00:00] [INFO] [printer.go:56] ["loaded config"] [config="{\"host\":\"0.0.0.0\",\"advertise-address\":\"10.155.111.136\",\"port\":4000,\"cors\":\"\",\"store\":\"tikv\",\"path\":\"10.155.111.139:2379,10.155.111.136:2379,10.155.111.236:2379\",\"socket\":\"\",\"lease\":\"45s\",\"run-ddl\":true,\"split-table\":true,\"token-limit\":1000,\"oom-use-tmp-storage\":true,\"tmp-storage-path\":\"/tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage\",\"oom-action\":\"log\",\"mem-quota-query\":1073741824,\"tmp-storage-quota\":-1,\"enable-streaming\":false,\"enable-batch-dml\":false,\"lower-case-table-names\":2,\"server-version\":\"\",\"log\":{\"level\":\"warn\",\"format\":\"text\",\"disable-timestamp\":null,\"enable-timestamp\":null,\"disable-error-stack\":null,\"enable-error-stack\":null,\"file\":{\"filename\":\"/home/tidb/deploy/log/tidb.log\",\"max-size\":300,\"max-days\":0,\"max-backups\":0},\"enable-slow-log\":true,\"slow-query-file\":\"/home/tidb/deploy/log/tidb_slow_query.log\",\"slow-threshold\":300,\"expensive-threshold\":10000,\"query-log-max-len\":4096,\"record-plan-in-slow-log\":1},\"security\":{\"skip-grant-table\":false,\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"require-secure-transport\":false,\"cluster-ssl-ca\":\"\",\"cluster-ssl-cert\":\"\",\"cluster-ssl-key\":\"\",\"cluster-verify-cn\":null},\"status\":{\"status-host\":\"0.0.0.0\",\"metrics-addr\":\"\",\"status-port\":10080,\"metrics-interval\":15,\"report-status\":true,\"record-db-qps\":false},\"performance\":{\"max-procs\":0,\"max-memory\":0,\"stats-lease\":\"3s\",\"stmt-count-limit\":5000,\"feedback-probability\":0.05,\"query-feedback-limit\":1024,\"pseudo-estimate-ratio\":0.8,\"force-priority\":\"NO_PRIORITY\",\"bind-info-lease\":\"3s\",\"txn-total-size-limit\":104857600,\"tcp-keep-alive\":true,\"cross-join\":true,\"run-auto-analyze\":true,\"agg-push-down-join\":false,\"committer-concurrency\":16,\"max-txn-ttl\":600000},\"prepared-plan-cache\":{\"enabled\":false,\"capacity\":100,\"memory-guard-ratio\":0.1},\"opentracing\":{\"enable\":false,\"rpc-metrics\":false,\"sampler\":{\"type\":\"const\",\"param\":1,\"sampling-server-url\":\"\",\"max-operations\":0,\"sampling-refresh-interval\":0},\"reporter\":{\"queue-size\":0,\"buffer-flush-interval\":0,\"log-spans\":false,\"local-agent-host-port\":\"\"}},\"proxy-protocol\":{\"networks\":\"\",\"header-timeout\":5},\"tikv-client\":{\"grpc-connection-count\":4,\"grpc-keepalive-time\":10,\"grpc-keepalive-timeout\":3,\"commit-timeout\":\"41s\",\"max-batch-size\":128,\"overload-threshold\":200,\"max-batch-wait-time\":0,\"batch-wait-size\":8,\"enable-chunk-rpc\":true,\"region-cache-ttl\":600,\"store-limit\":0,\"store-liveness-timeout\":\"120s\",\"copr-cache\":{\"enabled\":true,\"capacity-mb\":1000,\"admission-max-result-mb\":10,\"admission-min-process-ms\":5}},\"binlog\":{\"enable\":false,\"ignore-error\":false,\"write-timeout\":\"15s\",\"binlog-socket\":\"\",\"strategy\":\"range\"},\"compatible-kill-query\":false,\"plugin\":{\"dir\":\"\",\"load\":\"\"},\"pessimistic-txn\":{\"enable\":true,\"max-retry-count\":256},\"check-mb4-value-in-utf8\":true,\"max-index-length\":3072,\"alter-primary-key\":false,\"treat-old-version-utf8-as-utf8mb4\":true,\"enable-table-lock\":false,\"delay-clean-table-lock\":0,\"split-region-max-num\":1000,\"stmt-summary\":{\"enable\":true,\"enable-internal-query\":false,\"max-stmt-count\":200,\"max-sql-length\":4096,\"refresh-interval\":1800,\"history-size\":24},\"repair-mode\":false,\"repair-table-list\":[],\"isolation-read\":{\"engines\":[\"tikv\",\"tiflash\",\"tidb\"]},\"max-server-connections\":0,\"new_collations_enabled_on_first_bootstrap\":false,\"experimental\":{\"allow-auto-random\":false,\"allow-expression-index\":false}}"]

[2020/05/27 08:33:23.573 +00:00] [FATAL] [terror.go:348] ["unexpected error"] [error="[ddl:1273]Unknown collation: 'utf8mb4'"] [stack="github.com/pingcap/log.Fatal\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/pkg/mod/github.com/pingcap/log@v0.0.0-20200117041106-d28c14d3b1cd/global.go:59\ngithub.com/pingcap/parser/terror.MustNil\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20200427031542-879c7bd4f27d/terror/terror.go:348\nmain.createStoreAndDomain\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:296\nmain.main\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:181\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"]

**Comment by Still4** [comment]
mysql> select distinct TABLE_COLLATION from TABLES;
+--------------------+
| TABLE_COLLATION    |
+--------------------+
| utf8_general_ci    | 
| utf8mb4_0900_ai_ci | 
| utf8mb4_bin        | 
| utf8_bin           | 
| utf8mb4_general_ci | 
+--------------------+

**Comment by Still4** [comment]
tail tidb_stderr.log

config file conf/tidb.toml contained unknown configuration options: txn-local-latches
config file conf/tidb.toml contained unknown configuration options: txn-local-latches
config file conf/tidb.toml contained unknown configuration options: txn-local-latches
config file conf/tidb.toml contained unknown configuration options: txn-local-latches
config file conf/tidb.toml contained unknown configuration options: txn-local-latches

**Comment by wjhuang2016** [comment]
@Still4 
Hi, what's the version of the cluster first initialized?

**Comment by bb7133** [comment]
> mysql> select distinct TABLE_COLLATION from TABLES;
> +--------------------+
> | TABLE_COLLATION |
> +--------------------+
> | utf8_general_ci |
> | utf8mb4_0900_ai_ci |
> | utf8mb4_bin |
> | utf8_bin |
> | utf8mb4_general_ci |
> +--------------------+

Hi, is your TiDB able to work now? Or you just downgraded the cluster back to 3.0.9?

**Comment by Still4** [comment]
> @Still4
> Hi, what's the version of the cluster first initialized?

date should be around 2018-04, version should be v1.0.6

last night db upgraded to v3.1.1, kv was upgraded to 4.0.0-rc last month

KV 
```
Release Version:   4.0.0-rc
Git Commit Hash:   f45d0c963df3ee4b1011caf5eb146cacd1fbbad8
Git Commit Branch: heads/refs/tags/v4.0.0-rc
UTC Build Time:    2020-04-08 07:25:16
Rust Version:      rustc 1.42.0-nightly (0de96d37f 2019-12-19)
Enable Features:   jemalloc portable sse protobuf-codec
Profile:           dist_release
```

DB
```
Release Version: v3.1.1
Git Commit Hash: df4c9f05c49c6b8de74083d3befc20ecd8cefbfd
Git Branch: heads/refs/tags/v3.1.1
UTC Build Time: 2020-04-30 10:25:16
GoVersion: go version go1.13 linux/amd64
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

**Comment by Still4** [comment]
tidb is working fine right now, just cannot upgrade db nodes

**Comment by wjhuang2016** [comment]
@Still4 
Could you execute the following command and show us the result?
`show create table mysql.tidb;`

**Comment by Still4** [comment]
@wjhuang2016 
CREATE TABLE `tidb` (
  `VARIABLE_NAME` varchar(64) NOT NULL,
  `VARIABLE_VALUE` varchar(1024) DEFAULT NULL,
  `COMMENT` varchar(1024) DEFAULT NULL,
  PRIMARY KEY (`VARIABLE_NAME`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

**Comment by wjhuang2016** [comment]
@Still4 
Please try this again:
`select distinct(COLLATION_NAME) from information_schema.Columns;`

**Comment by Still4** [comment]
@wjhuang2016 
mysql> select COLLATION_NAME,count(1) from information_schema.Columns group by COLLATION_NAME order by COLLATION_NAME;
+--------------------+----------+
| COLLATION_NAME     | count(1) |
+--------------------+----------+
| NULL               |     7074 | 
| binary             |       16 | 
| utf8_bin           |       35 | 
| utf8_unicode_ci    |       16 | 
| utf8mb4_0900_ai_ci |        3 | 
| utf8mb4_bin        |    11443 | 
| utf8mb4_general_ci |        6 | 
+--------------------+----------+
7 rows in set (0.08 sec)

**Comment by wjhuang2016** [comment]
@Still4 
Please try this again:
`select DEFAULT_COLLATION_NAME from information_schema.schemata;`

**Comment by Still4** [comment]
@wjhuang2016 
```
mysql> select DEFAULT_COLLATION_NAME from information_schema.schemata;
+------------------------+
| DEFAULT_COLLATION_NAME |
+------------------------+
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_general_ci     | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_general_ci     | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_general_ci     | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_general_ci     | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
| utf8mb4_bin            | 
+------------------------+
48 rows in set (0.00 sec)
```

**Comment by wjhuang2016** [comment]
@Still4 
https://drive.google.com/file/d/14o_rnc8pyeqVwozra1PPtEbS4H8JneIx/view?usp=sharing
Could you try this:
./tidb-server --pd={IP:PORT} --store="tikv"
Then share the TiDB log to us?

**Comment by Still4** [comment]
@wjhuang2016 
./tidb-server --pd={10.155.111.139:3000} --store="tikv"
flag provided but not defined: -pd

**Comment by wjhuang2016** [comment]
@Still4 
Sorry.
`./tidb-server -path {IP:PORT} -store "tikv"`

**Comment by wjhuang2016** [comment]
BTW, the port of PD is 2379 by default.

**Comment by Still4** [comment]
@wjhuang2016 
```
[tidb@181-139 bin]$ ./tidb-server --path 10.155.111.139:2379 --store="tikv"
[2020/05/28 03:32:33.319 -05:00] [INFO] [printer.go:42] ["Welcome to TiDB."] ["Release Version"=v4.0.0-rc.2] [Edition=Community] ["Git Commit Hash"=dfbb1ff8b0a8d55893bcea652748c40b9a9b91ad] ["Git Branch"=heads/refs/tags/v4.0.0-rc.2] ["UTC Build Time"="2020-05-15 11:54:25"] [GoVersion=go1.13] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=v3.0.0-60965b006877ca7234adaced7890d7b029ed1306]
[2020/05/28 03:32:33.320 -05:00] [INFO] [printer.go:56] ["loaded config"] [config="{\"host\":\"0.0.0.0\",\"advertise-address\":\"0.0.0.0\",\"port\":4000,\"cors\":\"\",\"store\":\"tikv\",\"path\":\"10.155.111.139:2379\",\"socket\":\"\",\"lease\":\"45s\",\"run-ddl\":true,\"split-table\":true,\"token-limit\":1000,\"oom-use-tmp-storage\":true,\"tmp-storage-path\":\"/tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage\",\"oom-action\":\"cancel\",\"mem-quota-query\":1073741824,\"tmp-storage-quota\":-1,\"enable-streaming\":false,\"enable-batch-dml\":false,\"lower-case-table-names\":2,\"server-version\":\"\",\"log\":{\"level\":\"info\",\"format\":\"text\",\"disable-timestamp\":null,\"enable-timestamp\":null,\"disable-error-stack\":null,\"enable-error-stack\":null,\"file\":{\"filename\":\"\",\"max-size\":300,\"max-days\":0,\"max-backups\":0},\"enable-slow-log\":true,\"slow-query-file\":\"tidb-slow.log\",\"slow-threshold\":300,\"expensive-threshold\":10000,\"query-log-max-len\":4096,\"record-plan-in-slow-log\":1},\"security\":{\"skip-grant-table\":false,\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"require-secure-transport\":false,\"cluster-ssl-ca\":\"\",\"cluster-ssl-cert\":\"\",\"cluster-ssl-key\":\"\",\"cluster-verify-cn\":null},\"status\":{\"status-host\":\"0.0.0.0\",\"metrics-addr\":\"\",\"status-port\":10080,\"metrics-interval\":15,\"report-status\":true,\"record-db-qps\":false},\"performance\":{\"max-procs\":0,\"max-memory\":0,\"stats-lease\":\"3s\",\"stmt-count-limit\":5000,\"feedback-probability\":0.05,\"query-feedback-limit\":1024,\"pseudo-estimate-ratio\":0.8,\"force-priority\":\"NO_PRIORITY\",\"bind-info-lease\":\"3s\",\"txn-total-size-limit\":104857600,\"tcp-keep-alive\":true,\"cross-join\":true,\"run-auto-analyze\":true,\"agg-push-down-join\":false,\"committer-concurrency\":16,\"max-txn-ttl\":600000},\"prepared-plan-cache\":{\"enabled\":false,\"capacity\":100,\"memory-guard-ratio\":0.1},\"opentracing\":{\"enable\":false,\"rpc-metrics\":false,\"sampler\":{\"type\":\"const\",\"param\":1,\"sampling-server-url\":\"\",\"max-operations\":0,\"sampling-refresh-interval\":0},\"reporter\":{\"queue-size\":0,\"buffer-flush-interval\":0,\"log-spans\":false,\"local-agent-host-port\":\"\"}},\"proxy-protocol\":{\"networks\":\"\",\"header-timeout\":5},\"tikv-client\":{\"grpc-connection-count\":4,\"grpc-keepalive-time\":10,\"grpc-keepalive-timeout\":3,\"commit-timeout\":\"41s\",\"max-batch-size\":128,\"overload-threshold\":200,\"max-batch-wait-time\":0,\"batch-wait-size\":8,\"enable-chunk-rpc\":true,\"region-cache-ttl\":600,\"store-limit\":0,\"store-liveness-timeout\":\"120s\",\"copr-cache\":{\"enabled\":true,\"capacity-mb\":1000,\"admission-max-result-mb\":10,\"admission-min-process-ms\":5}},\"binlog\":{\"enable\":false,\"ignore-error\":false,\"write-timeout\":\"15s\",\"binlog-socket\":\"\",\"strategy\":\"range\"},\"compatible-kill-query\":false,\"plugin\":{\"dir\":\"\",\"load\":\"\"},\"pessimistic-txn\":{\"enable\":true,\"max-retry-count\":256},\"check-mb4-value-in-utf8\":true,\"max-index-length\":3072,\"alter-primary-key\":false,\"treat-old-version-utf8-as-utf8mb4\":true,\"enable-table-lock\":false,\"delay-clean-table-lock\":0,\"split-region-max-num\":1000,\"stmt-summary\":{\"enable\":true,\"enable-internal-query\":false,\"max-stmt-count\":200,\"max-sql-length\":4096,\"refresh-interval\":1800,\"history-size\":24},\"repair-mode\":false,\"repair-table-list\":[],\"isolation-read\":{\"engines\":[\"tikv\",\"tiflash\",\"tidb\"]},\"max-server-connections\":0,\"new_collations_enabled_on_first_bootstrap\":false,\"experimental\":{\"allow-auto-random\":false,\"allow-expression-index\":false}}"]
[2020/05/28 03:32:33.320 -05:00] [INFO] [main.go:341] ["disable Prometheus push client"]
[2020/05/28 03:32:33.320 -05:00] [INFO] [store.go:68] ["new store"] [path=tikv://10.155.111.139:2379]
[2020/05/28 03:32:33.320 -05:00] [INFO] [systime_mon.go:25] ["start system time monitor"]
[2020/05/28 03:32:33.320 -05:00] [INFO] [client.go:141] ["[pd] create pd client with endpoints"] [pd-address="[10.155.111.139:2379]"]
[2020/05/28 03:32:33.324 -05:00] [INFO] [base_client.go:226] ["[pd] update member urls"] [old-urls="[http://10.155.111.139:2379]"] [new-urls="[http://10.155.111.136:2379,http://10.155.111.139:2379,http://10.155.111.236:2379]"]
[2020/05/28 03:32:33.324 -05:00] [INFO] [base_client.go:242] ["[pd] switch leader"] [new-leader=http://10.155.111.139:2379] [old-leader=]
[2020/05/28 03:32:33.324 -05:00] [INFO] [base_client.go:92] ["[pd] init cluster id"] [cluster-id=6513119302746624016]
[2020/05/28 03:32:33.326 -05:00] [INFO] [store.go:74] ["new store with retry success"]
[2020/05/28 03:32:33.329 -05:00] [INFO] [tidb.go:72] ["new domain"] [store=tikv-6513119302746624016] ["ddl lease"=45s] ["stats lease"=3s]
[2020/05/28 03:32:33.333 -05:00] [INFO] [ddl.go:321] ["[ddl] start DDL"] [ID=651c0836-6d87-441c-a341-4e57fac1ceae] [runWorker=true]
[2020/05/28 03:32:33.335 -05:00] [INFO] [ddl.go:309] ["[ddl] start delRangeManager OK"] ["is a emulator"=false]
[2020/05/28 03:32:33.335 -05:00] [INFO] [ddl_worker.go:130] ["[ddl] start DDL worker"] [worker="worker 1, tp general"]
[2020/05/28 03:32:33.335 -05:00] [INFO] [ddl_worker.go:130] ["[ddl] start DDL worker"] [worker="worker 2, tp add index"]
[2020/05/28 03:32:33.479 -05:00] [INFO] [domain.go:144] ["full load InfoSchema success"] [usedSchemaVersion=0] [neededSchemaVersion=23415] ["start time"=134.260063ms]
[2020/05/28 03:32:33.480 -05:00] [INFO] [domain.go:368] ["full load and reset schema validator"]
[2020/05/28 03:32:33.484 -05:00] [FATAL] [terror.go:348] ["unexpected error"] [error="[ddl:1273]Unknown collation: 'utf8mb4'"] [stack="github.com/pingcap/log.Fatal\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/pkg/mod/github.com/pingcap/log@v0.0.0-20200117041106-d28c14d3b1cd/global.go:59\ngithub.com/pingcap/parser/terror.MustNil\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20200427031542-879c7bd4f27d/terror/terror.go:348\nmain.createStoreAndDomain\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:296\nmain.main\n\t/home/jenkins/agent/workspace/tidb_v4.0.0-rc.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:181\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"]
```

**Comment by wjhuang2016** [comment]
@Still4 
Did you the use TiDB binary I posted? https://drive.google.com/file/d/14o_rnc8pyeqVwozra1PPtEbS4H8JneIx/view?usp=sharing

**Comment by Still4** [comment]
@wjhuang2016 
sry, I thought that's a folder to upload log, haha

**Comment by Still4** [comment]
@wjhuang2016 

```
[tidb@181-139 bin]$ ./tidb-server --path 10.155.111.139:2379 --store="tikv"
[2020/05/28 05:36:49.360 -05:00] [INFO] [printer.go:42] ["Welcome to TiDB."] ["Release Version"=v4.0.0-rc.2-1-g9482f612e] [Edition=Community] ["Git Commit Hash"=9482f612e350a0f9dcbb5013f6698f3d7337345d] ["Git Branch"=rc2_colltion_hotfix] ["UTC Build Time"="2020-05-28 07:45:04"] [GoVersion=go1.13] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=v3.0.0-60965b006877ca7234adaced7890d7b029ed1306]
[2020/05/28 05:36:49.361 -05:00] [INFO] [printer.go:56] ["loaded config"] [config="{\"host\":\"0.0.0.0\",\"advertise-address\":\"0.0.0.0\",\"port\":4000,\"cors\":\"\",\"store\":\"tikv\",\"path\":\"10.155.111.139:2379\",\"socket\":\"\",\"lease\":\"45s\",\"run-ddl\":true,\"split-table\":true,\"token-limit\":1000,\"oom-use-tmp-storage\":true,\"tmp-storage-path\":\"/tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage\",\"oom-action\":\"cancel\",\"mem-quota-query\":1073741824,\"tmp-storage-quota\":-1,\"enable-streaming\":false,\"enable-batch-dml\":false,\"lower-case-table-names\":2,\"server-version\":\"\",\"log\":{\"level\":\"info\",\"format\":\"text\",\"disable-timestamp\":null,\"enable-timestamp\":null,\"disable-error-stack\":null,\"enable-error-stack\":null,\"file\":{\"filename\":\"\",\"max-size\":300,\"max-days\":0,\"max-backups\":0},\"enable-slow-log\":true,\"slow-query-file\":\"tidb-slow.log\",\"slow-threshold\":300,\"expensive-threshold\":10000,\"query-log-max-len\":4096,\"record-plan-in-slow-log\":1},\"security\":{\"skip-grant-table\":false,\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"require-secure-transport\":false,\"cluster-ssl-ca\":\"\",\"cluster-ssl-cert\":\"\",\"cluster-ssl-key\":\"\",\"cluster-verify-cn\":null},\"status\":{\"status-host\":\"0.0.0.0\",\"metrics-addr\":\"\",\"status-port\":10080,\"metrics-interval\":15,\"report-status\":true,\"record-db-qps\":false},\"performance\":{\"max-procs\":0,\"max-memory\":0,\"stats-lease\":\"3s\",\"stmt-count-limit\":5000,\"feedback-probability\":0.05,\"query-feedback-limit\":1024,\"pseudo-estimate-ratio\":0.8,\"force-priority\":\"NO_PRIORITY\",\"bind-info-lease\":\"3s\",\"txn-total-size-limit\":104857600,\"tcp-keep-alive\":true,\"cross-join\":true,\"run-auto-analyze\":true,\"agg-push-down-join\":false,\"committer-concurrency\":16,\"max-txn-ttl\":600000},\"prepared-plan-cache\":{\"enabled\":false,\"capacity\":100,\"memory-guard-ratio\":0.1},\"opentracing\":{\"enable\":false,\"rpc-metrics\":false,\"sampler\":{\"type\":\"const\",\"param\":1,\"sampling-server-url\":\"\",\"max-operations\":0,\"sampling-refresh-interval\":0},\"reporter\":{\"queue-size\":0,\"buffer-flush-interval\":0,\"log-spans\":false,\"local-agent-host-port\":\"\"}},\"proxy-protocol\":{\"networks\":\"\",\"header-timeout\":5},\"tikv-client\":{\"grpc-connection-count\":4,\"grpc-keepalive-time\":10,\"grpc-keepalive-timeout\":3,\"commit-timeout\":\"41s\",\"max-batch-size\":128,\"overload-threshold\":200,\"max-batch-wait-time\":0,\"batch-wait-size\":8,\"enable-chunk-rpc\":true,\"region-cache-ttl\":600,\"store-limit\":0,\"store-liveness-timeout\":\"120s\",\"copr-cache\":{\"enabled\":true,\"capacity-mb\":1000,\"admission-max-result-mb\":10,\"admission-min-process-ms\":5}},\"binlog\":{\"enable\":false,\"ignore-error\":false,\"write-timeout\":\"15s\",\"binlog-socket\":\"\",\"strategy\":\"range\"},\"compatible-kill-query\":false,\"plugin\":{\"dir\":\"\",\"load\":\"\"},\"pessimistic-txn\":{\"enable\":true,\"max-retry-count\":256},\"check-mb4-value-in-utf8\":true,\"max-index-length\":3072,\"alter-primary-key\":false,\"treat-old-version-utf8-as-utf8mb4\":true,\"enable-table-lock\":false,\"delay-clean-table-lock\":0,\"split-region-max-num\":1000,\"stmt-summary\":{\"enable\":true,\"enable-internal-query\":false,\"max-stmt-count\":200,\"max-sql-length\":4096,\"refresh-interval\":1800,\"history-size\":24},\"repair-mode\":false,\"repair-table-list\":[],\"isolation-read\":{\"engines\":[\"tikv\",\"tiflash\",\"tidb\"]},\"max-server-connections\":0,\"new_collations_enabled_on_first_bootstrap\":false,\"experimental\":{\"allow-auto-random\":false,\"allow-expression-index\":false}}"]
[2020/05/28 05:36:49.362 -05:00] [INFO] [main.go:344] ["disable Prometheus push client"]
[2020/05/28 05:36:49.362 -05:00] [INFO] [store.go:68] ["new store"] [path=tikv://10.155.111.139:2379]
[2020/05/28 05:36:49.362 -05:00] [INFO] [client.go:141] ["[pd] create pd client with endpoints"] [pd-address="[10.155.111.139:2379]"]
[2020/05/28 05:36:49.362 -05:00] [INFO] [systime_mon.go:25] ["start system time monitor"]
[2020/05/28 05:36:49.365 -05:00] [INFO] [base_client.go:226] ["[pd] update member urls"] [old-urls="[http://10.155.111.139:2379]"] [new-urls="[http://10.155.111.136:2379,http://10.155.111.139:2379,http://10.155.111.236:2379]"]
[2020/05/28 05:36:49.365 -05:00] [INFO] [base_client.go:242] ["[pd] switch leader"] [new-leader=http://10.155.111.139:2379] [old-leader=]
[2020/05/28 05:36:49.365 -05:00] [INFO] [base_client.go:92] ["[pd] init cluster id"] [cluster-id=6513119302746624016]
[2020/05/28 05:36:49.366 -05:00] [INFO] [store.go:74] ["new store with retry success"]
[2020/05/28 05:36:49.370 -05:00] [INFO] [tidb.go:72] ["new domain"] [store=tikv-6513119302746624016] ["ddl lease"=45s] ["stats lease"=3s]
[2020/05/28 05:36:49.373 -05:00] [INFO] [ddl.go:321] ["[ddl] start DDL"] [ID=cd0e99da-7739-4695-ad59-40b5d26dabba] [runWorker=true]
[2020/05/28 05:36:49.375 -05:00] [INFO] [ddl.go:309] ["[ddl] start delRangeManager OK"] ["is a emulator"=false]
[2020/05/28 05:36:49.375 -05:00] [INFO] [ddl_worker.go:130] ["[ddl] start DDL worker"] [worker="worker 1, tp general"]
[2020/05/28 05:36:49.375 -05:00] [INFO] [ddl_worker.go:130] ["[ddl] start DDL worker"] [worker="worker 2, tp add index"]
[2020/05/28 05:36:49.531 -05:00] [INFO] [domain.go:144] ["full load InfoSchema success"] [usedSchemaVersion=0] [neededSchemaVersion=23437] ["start time"=147.936736ms]
[2020/05/28 05:36:49.532 -05:00] [INFO] [domain.go:368] ["full load and reset schema validator"]
[2020/05/28 05:36:49.536 -05:00] [WARN] [main.go:297] [fail] ["error stk"="[ddl:1273]Unknown collation: 'utf8mb4'\ngithub.com/pingcap/errors.AddStack\n\t/home/wj/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/parser/terror.(*Error).GenWithStackByArgs\n\t/home/wj/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/pingcap/parser@v0.0.0-20200427031542-879c7bd4f27d/terror/terror.go:243\ngithub.com/pingcap/parser/charset.GetCollationByName\n\t/home/wj/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/pingcap/parser@v0.0.0-20200427031542-879c7bd4f27d/charset/charset.go:175\ngithub.com/pingcap/tidb/util/collate.GetCollationByName\n\t/home/wj/tidb/util/collate/collate.go:174\ngithub.com/pingcap/tidb/sessionctx/variable.ValidateSetSystemVar\n\t/home/wj/tidb/sessionctx/variable/varsutil.go:761\ngithub.com/pingcap/tidb/sessionctx/variable.SetSessionSystemVar\n\t/home/wj/tidb/sessionctx/variable/varsutil.go:215\ngithub.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded\n\t/home/wj/tidb/session/session.go:2013\ngithub.com/pingcap/tidb/session.(*session).execute\n\t/home/wj/tidb/session/session.go:1098\ngithub.com/pingcap/tidb/session.(*session).Execute\n\t/home/wj/tidb/session/session.go:1089\ngithub.com/pingcap/tidb/session.loadParameter\n\t/home/wj/tidb/session/session.go:1620\ngithub.com/pingcap/tidb/session.loadSystemTZ\n\t/home/wj/tidb/session/session.go:1597\ngithub.com/pingcap/tidb/session.BootstrapSession\n\t/home/wj/tidb/session/session.go:1666\nmain.createStoreAndDomain\n\t/home/wj/tidb/tidb-server/main.go:295\nmain.main\n\t/home/wj/tidb/tidb-server/main.go:181\nruntime.main\n\t/home/wj/.gvm/gos/go1.13/src/runtime/proc.go:203\nruntime.goexit\n\t/home/wj/.gvm/gos/go1.13/src/runtime/asm_amd64.s:1357"]
[2020/05/28 05:36:49.536 -05:00] [FATAL] [terror.go:348] ["unexpected error"] [error="[ddl:1273]Unknown collation: 'utf8mb4'"] [stack="github.com/pingcap/log.Fatal\n\t/home/wj/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/pingcap/log@v0.0.0-20200117041106-d28c14d3b1cd/global.go:59\ngithub.com/pingcap/parser/terror.MustNil\n\t/home/wj/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/pingcap/parser@v0.0.0-20200427031542-879c7bd4f27d/terror/terror.go:348\nmain.createStoreAndDomain\n\t/home/wj/tidb/tidb-server/main.go:299\nmain.main\n\t/home/wj/tidb/tidb-server/main.go:181\nruntime.main\n\t/home/wj/.gvm/gos/go1.13/src/runtime/proc.go:203"]
```

**Comment by wjhuang2016** [comment]
@Still4 
Thanks. Please check the following variables:
`select @@global.collation_connection, @@collation_database, @@collation_server;`

**Comment by Still4** [comment]
@wjhuang2016 
mysql> select @@global.collation_connection, @@collation_database, @@collation_server;
+-------------------------------+----------------------+--------------------+
| @@global.collation_connection | @@collation_database | @@collation_server |
+-------------------------------+----------------------+--------------------+
| utf8mb4                       | utf8mb4_general_ci   | utf8mb4            | 
+-------------------------------+----------------------+--------------------+
1 row in set (0.00 sec)

**Comment by wjhuang2016** [comment]
@Still4 
Not sure why the `collation_connection` and `collation_server` are `utf8mb4`, set it to `utf8mb4_bin` will be fine.
`set @@global.collation_connection="utf8mb4_bin", @@global.collation_server="utf8mb4_bin";`

**Comment by Still4** [comment]
@wjhuang2016 
thanks a lot, things goes fine now.
I think collation was changed about a year ago, fixing something with collation https://github.com/pingcap/tidb/issues/9430#issuecomment-466866599

**Comment by wjhuang2016** [comment]
@Still4 
You are welcome. We will check the code later and try to prevent it from happening again.

**Comment by wjhuang2016** [comment]
BTW, 4.0 GA is released just now, it's better to upgrade to 4.0 GA instead of RC2.

**Comment by Still4** [comment]
OK, I've been waiting for that, haha
